soroban-env-host/src/auth.rs

```diff
@@ -1,3 +1,197 @@
+//!
+//! Moreover this subsystem is responsible (as will be discussed in detail
+//! below) with facilitating two different _directions_ for each of these tasks:
+//!
+//!   - Contracts can _require_ auth services provided by this module.
+//!   - Contracts can _provide_ auth services required by this module.
+//!
+//! And again, in both directions the "auth services" required or provided may
+//! be _either_ of authorization, authentication, or both.
+//!
+//! ## Authorization
+//!
+//! Authorization judgments are explicitly requested by smart contracts, and
+//! relate to an ambient authorization context that is maintained by this
+//! module. Specifically a contract calls the host (using `require_auth` or
+//! `require_auth_for_args`) informing the host that the contract code needs
+//! authorization before it continues. If the requirement matches the context
+//! the call succeeds, otherwise it fails (and the contract typically traps).
+//!
+//! This interface raises obvious questions:
+//!
+//!   - What type of "requirements" can the contract require?
+//!   - What are the contents of the authentication context?
+//!
+//! To answer either of these questions we first need to take a slight detour
+//! into the notion of an _address_.
+//!
+//! ## Addresses
+//!
+//! Addresses generalize and abstract over accounts (corresponding to public
+//! keys) and contracts. To understand why this is helpful, consider what it
+//! means to authorize some action.
+//!
+//! Authorization (in the real world, and blockchains) to perform some sensitive
+//! action (such as, say, transferring money) takes two fairly different forms.
+//! Either:
+//!
+//!   1. The action is labeled with _some authority's name_ who has to approve
+//!      of the action, in which case we need convincing evidence that they
+//!      approved. Think of this as the case where your name is on a bank
+//!      account. Obviously if you ask for money out of that account, and you
+//!      provide a signature or PIN as an authentic credential, the withdrawal
+//!      is authorized. You are the designated owner or authority for it.
+//!
+//!   2. The action is labeled with _some set of rules_ that need to hold for
+//!      the action to be authorized, in which case we evaluate the rules in
+//!      context, whatever they may be. Think of this as the case where, say,
+//!      there are daily transfer limits on an account to minimize risks. Here
+//!      the rules have nothing to do with authority in the sense of a person,
+//!      merely the evaluation of a given situation.
+//!
+//! Some authorization scenarios will combine cases, and sometimes case 2 will
+//! delegate back to case 1 ("the rules" are "check for a signature from X").
+//! The examples above are illustrative of cases with the strictest difference,
+//! but that difference has crucial practical details:
+//!
+//!   - In case 1, the presence of a _valid signature_ (authentic credentials)
```

Comment:
The term 'signature' (here and some other places) creates a false impression that multisig is not supported, which is not true.

Reply:
Ok. I'll clarify.